allennlp/data/dataset_readers/semantic_role_labelling.py

```diff
@@ -0,0 +1,228 @@
+         not have co-reference annotations, each line is represented with a "-".
+
+    Parameters
+    ----------
+    ontonotes_filename : ``str``
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We similarly use this for both the premise and the hypothesis.  See :class:`TokenIndexer`.
+
+    Return
+    ------
+    A ``Dataset`` of ``Instances`` for Semantic Role Labelling.
+
+    """
+    def __init__(self,
+                 ontonotes_filename: str,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._ontonotes_filename = ontonotes_filename
+        self._token_indexers = token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    def process_sentence(self,
+                         sentence: List[str],
+                         verbal_predicates: List[int],
+                         predicate_argument_labels: List[List[str]]) -> List[Instance]:
+        """
+        Parameters
+        ---------
+        sentence : List[str], required.
+            The tokenised sentence.
+        verbal_predicates : List[int], required.
+            The indexes of the verbal predicates in the
+            sentence which have an associated annotation.
+        predicate_argument_labels : List[List[str]], required.
+            A list of predicate argument labels, one for each verbal_predicate. The
+            internal lists are of length: len(sentence).
+
+        Return
+        ------
+        A list of Instances.
+
+        """
+        sentence_field = TextField(sentence, self._token_indexers)
+        if not verbal_predicates:
+            # Sentence contains no predicates.
+            tags = TagField(["O" for _ in sentence], sentence_field)
+            verb_indicator = IndexField(None, sentence_field)
+            instance = Instance(fields={"tokens": sentence_field, "verb_indicator": verb_indicator, "tags": tags})
+            return [instance]
+        else:
+            instances = []
+            for verb_index, annotation in zip(verbal_predicates, predicate_argument_labels):
+
+                tags = TagField(annotation, sentence_field)
+                verb_indicator = IndexField(verb_index, sentence_field)
+                instances.append(Instance(fields={"tokens": sentence_field,
+                                                  "verb_indicator": verb_indicator,
+                                                  "tags": tags}))
+            return instances
+
+    @overrides
+    def read(self):
+        instances = []
+
+        sentence = []
+        verbal_predicates = []
+        predicate_argument_labels = []
+        current_span_label = []
+
+        for root, _, files in os.walk(self._ontonotes_filename):
+            for file in files:
+                # These are a relic of the dataset pre-processing. Every file will be duplicated
```

Comment:
What pre-processing are you referring to?  I don't see any reference to this anywhere else in the code.  Add some info about this to the class docstring, including the assumptions you make about how the directory you're walking is structured.